### PR TITLE
ROX-23835, ROX-26239: Mask graph deployment IDs for baseline compare 

### PR DESCRIPTION
**Pull request by rhybrillou**
### Description

Network graph masking alters the entities in network flows when these (entities) are not allowed access to by the requester roles (more precisely their access scope).

This becomes problematic when flows should be compared to the network baseline, as the entity IDs are masked as well.

This PR aims at relaxing the masking rules in order to allow flow comparison to the network baseline (deterministically computed masked ID so rules can be applied on baseline comparison as well).

### User-facing documentation

- [x] CHANGELOG ~~is updated **OR**~~ _update is not needed_
- [x] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) ~~is created and is linked above **OR**~~ _is not needed_

### Testing and quality

- [x] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [ ] CI results are inspected

#### Automated testing

The existing test coverage should be enough.

#### How I validated my change

Deploy

- create a restricted scope (e.g. stackrox namespace only), an admin role with this restricted scope, and an API token for that role
- check the network graph for the stackrox namespace on the deployed cluster
- capture the deployment name and IDs in the stackrox namespace
- capture the entity IDs in the network graph, as well as the entities communicating with the kube-dns deployment
- capture the request for network baseline for one of the deployments communicating with kube-dns
- ensure the flow between the deployment of the previous step and kube-dns is part of the network baseline
- capture the network graph rendered for the API token with restricted scope
- replay the request for network baseline using the entity IDs that were exposed in the network graph with restricted scope
- ensure the flow to the masked kube-dns deployment is flagged as BASELINE despite its target being masked

### CODE REVIEW

File: central/networkbaseline/service/service_impl.go [hunk 1]
```diff
@@ -161,11 +162,24 @@ func (s *serviceImpl) getBaselinePeerByEntityID(
...
+		// Insert second peer entry for the masked deployment ID
+		// that can be issued by the network graph deployment masker
+		if peerType == storage.NetworkEntityInfo_DEPLOYMENT {
+			deploymentName := peer.GetEntity().GetInfo().GetDeployment().GetName()
+			maskedID := uuid.NewV5FromNonUUIDs(peerId, deploymentName).String()
+			maskedKey := networkgraph.Entity{
+				Type: peerType,
+				ID:   maskedID,
+			}
+			result[maskedKey] = peer
```
Comment (vikin91) [inline central/networkbaseline/service/service_impl.go:181]: So we are adding it twice to the `result` map?

I am lacking context of what this map precisely does, so I would appreciate some explanation to get more context.

File: central/networkbaseline/service/service_impl.go [hunk 2]
```diff
@@ -161,11 +162,24 @@ func (s *serviceImpl) getBaselinePeerByEntityID(
...
+		peerId := peer.GetEntity().GetInfo().GetId()
 		key := networkgraph.Entity{
-			Type: peer.GetEntity().GetInfo().GetType(),
-			ID:   peer.GetEntity().GetInfo().GetId(),
+			Type: peerType,
+			ID:   peerId,
 		}
 		result[key] = peer
+		// Insert second peer entry for the masked deployment ID
+		// that can be issued by the network graph deployment masker
```
Comment (vikin91) [inline central/networkbaseline/service/service_impl.go:173]: This was a tricky issue to investigate and I remember Luis spending some time on it. Could we maybe add some additional info to the comment to explain why are we doing this deterministic masking?

**Review by vikin91 — COMMENTED** [review]
The solution looks elegant and compact. I left two comments, but I have not verified that this solution works, as Meriln is requested for review. If you'd like someone from Maple to review that properly the please let us know (however, this may take a moment, as Luis is off and we would need to get some more context on that area of code).

File: central/networkbaseline/service/service_impl.go [hunk 3]
```diff
@@ -154,23 +155,48 @@ func (s *serviceImpl) getStatusesForPeers(
...
+		}
 	}
 
 	return result
 }
 
+func getMaskedDeploymentID(id string, name string) string {
+	return uuid.NewV5FromNonUUIDs(id, name).String()
+}
+
```
Comment (lvalerom) [inline central/networkbaseline/service/service_impl.go:199]: [nit] Could be merge this function with the one defined in `masker.go`?

**Review by lvalerom — APPROVED** [review]
The change looks good to me.

**Pull request merged**